core/Archive/ArchiveInvalidator.php

```diff
@@ -228,16 +238,30 @@ private function deleteOptionLike($id)
 
     /**
      * @param $idSites int[]
-     * @param $dates Date[]
+     * @param $dates Date[]|string[]
      * @param $period string
      * @param $segment Segment
      * @param bool $cascadeDown
+     * @param bool $forceInvalidateNonexistantRanges set true to force inserting rows for ranges in archive_invalidations
+     * @param string $name null to make sure every plugin is archived when this invalidation is processed by core:archive,
+     *                     or a plugin name to only archive the specific plugin.
      * @return InvalidationResult
      * @throws \Exception
      */
     public function markArchivesAsInvalidated(array $idSites, array $dates, $period, Segment $segment = null, $cascadeDown = false,
-                                              $forceInvalidateNonexistantRanges = false)
+                                              $forceInvalidateNonexistantRanges = false, $name = null)
     {
+        $plugin = null;
+        if ($name && strpos($name, '.') !== false) {
+            list($plugin) = explode('.', $name);
+        }
+
+        if ($plugin
+            && !Manager::getInstance()->isPluginActivated($plugin)
+        ) {
+            throw new \Exception("Plugin is not activated: '$plugin'");
```

Comment:
This specific case shouldn't be an issue, since once the entries are in archive_invalidations, we don't try to invalidate for the plugin.

If the exception could occur during core:archive then the command would fail... but CronArchive doesn't try to invalidate a specific plugin.

In your specific case, the remaining 900 would be pulled, then in CronArchive we would see the plugin is disabled, so we would skip and remove the invalidation entry.